### PR TITLE
Adjust output for NoEnvironmentVariablePropertyAnalyzer based on CustomConfuguration

### LINKED ISSUE #10429: [BuildCheck][FeatureRequest] Env variable BuildCheck reports value
Current implementation of the EnvVariable BuildCheck when turned on reports the actual value that were set during the build of the Environment variable. 
```
"Property is derived from environment variable: 'OS' with value: 'Windows_NT'. Properties should be passed explicitly using the /p option."
```
Howver in some scenarios it could be unexpected behavior hence the suggestion is to remove the showing it by default and enable this behavior by explicitly configuring it.

### PR DESCRIPTION
**Pull request by YuliiaKovalova**
Fixes #10429

### Context
Environment variables might contain secret information. It order to have it exposed, customers should specify a special configuration:

`build_check.BC0103.allow_displaying_environment_variable_value=true`

### Changes Made
Add respecting of `allow_displaying_environment_variable_value` configuration.

### Testing
Added e2e tests

### CODE REVIEW

File: src/Build/BuildCheck/API/BuildAnalyzerRule.cs [hunk 1]
```diff
@@ -45,7 +49,7 @@ public class BuildAnalyzerRule
     /// <summary>
     /// Message format that will be used by the actual reports (<see cref="BuildCheckResult"/>) - those will just supply the actual arguments.
     /// </summary>
-    public string MessageFormat { get; }
+    public string MessageFormat { get; set; }
```
Comment (YuliiaKovalova) [inline src/Build/BuildCheck/API/BuildAnalyzerRule.cs:52]: @JanKrivanek , heads up on the change

File: src/Build/BuildCheck/Analyzers/NoEnvironmentVariablePropertyAnalyzer.cs [hunk 2]
```diff
@@ -9,25 +9,39 @@ namespace Microsoft.Build.Experimental.BuildCheck.Analyzers;
...
     public override IReadOnlyList<BuildAnalyzerRule> SupportedRules { get; } = [SupportedRule];
 
     public override void Initialize(ConfigurationContext configurationContext)
     {
-        // No custom configuration
+        foreach (CustomConfigurationData customConfigurationData in configurationContext.CustomConfigurationData)
+        {
+            bool? isVerboseEnvVarOutput = GetVerboseEnvVarOutputConfig(customConfigurationData, RuleId);
+            _isVerboseEnvVarOutput = isVerboseEnvVarOutput.HasValue && isVerboseEnvVarOutput.Value;           
+        }
```
Comment (JanKrivanek) [inline src/Build/BuildCheck/Analyzers/NoEnvironmentVariablePropertyAnalyzer.cs:44]: Your analyzer has just single rule - so you can as well call just `GetVerboseEnvVarOutputConfig(configurationContext.CustomConfigurationData[0], RuleId);`

Comment (JanKrivanek) [reply src/Build/BuildCheck/Analyzers/NoEnvironmentVariablePropertyAnalyzer.cs:44]: But your code is fine as well and actually more universal - so let's maybe keep it :-)

File: src/Build/BuildCheck/API/BuildAnalyzerRule.cs (hunk 1 above)
Comment (JanKrivanek) [reply src/Build/BuildCheck/API/BuildAnalyzerRule.cs:52]: I like the nicely readable and organized formats per both verbosity options. On the other hand adding a mutability to this class feels to high tradeoff.
Let's keep this immutable at the cost of less nice format (can be '{0}'). We can figure out better approach later onif needed often.

**Review by JanKrivanek — APPROVED** [review]
Looks good - once the Rule class is again immutable

**Review by surayya-MS — APPROVED** [review]
Looks good to me!

**Pull request merged**